layersvt/api_dump.h

```diff
@@ -833,389 +851,359 @@ class ApiDumpInstance {
         settings.stream() << "NULL";
     else
         settings.stream() << "\"" << object << "\"";
-    return settings.stream() << "</div>";
+    settings.stream() << "</div>";
 }
 
-inline std::ostream &dump_html_void(const void *object, const ApiDumpSettings &settings, int indents) {
+void dump_html_void(const void *object, const ApiDumpSettings &settings, int indents) {
     settings.stream() << "<div class='val'>";
-    OutputAddress(settings, object, false);
-    return settings.stream() << "</div>";
+    OutputAddress(settings, object);
+    settings.stream() << "</div>";
 }
 
-inline std::ostream &dump_html_int(int object, const ApiDumpSettings &settings, int indents) {
+void dump_html_int(int object, const ApiDumpSettings &settings, int indents) {
     settings.stream() << "<div class='val'>";
     settings.stream() << object;
-    return settings.stream() << "</div>";
+    settings.stream() << "</div>";
 }
 
-template <typename T, typename... Args>
-inline void dump_html_pNext(const T *object, const ApiDumpSettings &settings, const char *type_string, int indents,
-                            std::ostream &(*dump)(const T &, const ApiDumpSettings &, int, Args... args), Args... args) {
+template <typename T>
+void dump_html_pNext(const T *object, const ApiDumpSettings &settings, const char *type_string, int indents,
+                     void (*dump)(const T &, const ApiDumpSettings &, int)) {
     if (object == NULL) {
         settings.stream() << "<details class='data'><summary>";
         dump_html_nametype(settings.stream(), settings.showType(), "pNext", type_string);
         settings.stream() << "<div class='val'>NULL</div></summary></details>";
     } else {
-        dump_html_value(*object, settings, type_string, "pNext", indents, dump, args...);
+        dump_html_value(*object, settings, type_string, "pNext", indents, dump);
     }
 }
 
 //==================================== Json Backend Helpers ======================================//
 
-template <typename T, typename... Args>
-inline void dump_json_array(const T *array, size_t len, const ApiDumpSettings &settings, const char *type_string,
-                            const char *child_type, const char *name, int indents,
-                            std::ostream &(*dump)(const T, const ApiDumpSettings &, int, Args... args), Args... args) {
+void dump_json_function_head(ApiDumpInstance &dump_inst, const char *funcName, const char *funcReturn) {
+    const ApiDumpSettings &settings(dump_inst.settings());
+
+    if (!dump_inst.firstFunctionCallOnFrame()) settings.stream() << ",\n";
+
+    // Display api call name
+    settings.stream() << settings.indentation(2) << "{\n";
+    settings.stream() << settings.indentation(3) << "\"name\" : \"" << funcName << "\",\n";
+
+    // Display thread info
+    if (settings.showThreadAndFrame()) {
+        { settings.stream() << settings.indentation(3) << "\"thread\" : \"Thread " << dump_inst.threadID() << "\",\n"; }
```

Comment:
does this also emit the frame?

Reply:
The showThreadAndFrame refers to the current thread and current frame number. Both HTML and JSON output encode the 'frame' number in the parent objects, whereas what this is printing is inside the child object.